### PR TITLE
[AIC-py][editor] get endpoint: list_models

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[AIC-py][editor] get endpoint: list_models

- Name post response better (w. aiconfig)
- add get method (list models)

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/594).
* #595
* __->__ #594

### CODE REVIEW

File: python/src/aiconfig/editor/server/server.py [hunk 1]
```diff
@@ -122,42 +124,48 @@ async def run() -> FlaskPostResponse:
...
+        return HttpResponseWithAIConfig(
             message=f"Failed to add prompt: {err}",
             code=400,
             aiconfig=None,
         ).to_flask_format()
 
 
+@app.route("/api/list_models", methods=["GET"])
+def list_models() -> FlaskResponse:
+    out: list[str] = ModelParserRegistry.parser_ids()  # type: ignore
```
Comment (rholinshead) [inline python/src/aiconfig/editor/server/server.py:164]: Is this the same as `ModelParserRegistry.display_parsers().keys()`? parser ids vs model names is still a bit confusing to me.

Also, can we support searching by substring here? 

I had done:
```
request_json = request.get_json()
all_models: list[core_utils.JSONValue] = list(ModelParserRegistry.display_parsers().keys())
search = request_json.get("search", "")
models = [m for m in all_models if search in m]
```
which is currently erroring on bad request format but the general concept should be there (and I can scrap it in favour of your much better python code)

Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:164]: > parser ids vs model names is still a bit confusing to me.

Me too, dont know off the top of my head. @Ankush-lastmile ?

Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:164]: Ok just checked, this returns the mapping keys, which are the model names.

**Review by rholinshead — APPROVED** [review]
LGTM. Discussed search filtering offline. For now we'll do that client-side. For gradio workbook we can consider server side based on hugging face models

**Pull request merged**